cmake/sanitizer.cmake

```diff
@@ -0,0 +1,65 @@
+include(CheckCXXCompilerFlag)
+include(CheckCXXSourceRuns)
+
+set(ALLOWED_BUILD_TYPES Debug Release RelWithDebInfo MinSizeRel)
+set(ALL_SAN_FLAGS "")
+
+# Thread sanitizer
+set(THREAD_SAN_FLAGS "-fsanitize=thread")
+set(PREV_FLAG ${CMAKE_REQUIRED_FLAGS})
+set(CMAKE_REQUIRED_FLAGS "${THREAD_SAN_FLAGS}")
+check_cxx_source_runs("int main() { return 0; }" THREAD_SANITIZER_AVAILABLE)
+set(CMAKE_REQUIRED_FLAGS ${PREV_FLAG})
+if(THREAD_SANITIZER_AVAILABLE)
+    list(APPEND ALLOWED_BUILD_TYPES ThreadSan)
+    # Do not add Thread sanitizer to all sanitizer because it is incompatible with other sanitizer
```

Comment:
Since thread sanitizer are incompatible with other sanitizers, could you please create an additional CI run for those?

 `Thread` should be lower case here :wink:.